UKHomeOffice/sgmr-service
# Cg 196 handle signin redirect

TO TEST

Recommend 
-  clear cache/cookies
- `rm -rf node_modules package-lock.json`
- `npm i && npm run build-dev && npm start`

——

**You should be forced to sign in if you do not have a valid token**

NAV:

- Sign in
- Delete your token
- Click on a menu link (vessels, people, account)
- > you should be taken to the sign in page
- Sign in
- > you should be taken to the page of the menu link you clicked on


- Go to /reports
- Start a new report
- Go to page 2 (arrivals)
- Delete your token
- Click on save and continue
- > you should be taken to the sign in page
- Sign in
- > you should be taken back to page 2 (arrivals)
- Click back
- > you should be taken to /reports (rather than /verify)

- Go to /vessels
- Go to save a vessel
- Delete your token
- Click exit without saving
- > you should be taken to the sign in page
- Sign in
- > you should be returned to /vessels

——

FORMS:

- Go to /vessels
- Go to save a vessel
- Enter some details
- Delete your token
- Click save and continue
- > you should be taken to the sign in page
- Sign in
- > you should be returned to your vessel form with data populated
- Click save and continue

- Go to /vessels
- Click on your saved vessel
- Change some details
- Delete your token
- Click save and continue
- > you should be taken to the sign in page
- Sign in
- > you should be returned to your vessel form with changes populated
- click save
-> you should be returned to /vessels with the changes saved

- People works the same as vessels, follow the same pattern as above

- Go to /reports
- Click on start now
- Enter some details on the page
- Delete your token
- Click save and continue
- > you should be taken to the sign in page
- Sign in
- > you should be returned to your page on the form

Repeat for 
- Page 2 arrivals
- Page 3 vessel 
- Page 5 responsible person
Note there is currently no page 4, this will be built soon

- Go to /reports
- Click on any of your voyages to go to the check details page
- Delete your token
- Click on Change for any section
- > you should be taken to sign in
- Sign in
- > you should be returned to your voyage form
NOTE accepts and submit and cancel buttons currently have no functionality and cannot be tested, they will be built soon
